src/eliminate_contiguous.cpp

```diff
@@ -93,15 +98,25 @@ void eliminate_contiguous::apply(module& m) const
                 }
                 else if(prev->can_eval())
                 {
-                    auto c = op::contiguous{};
-                    auto r = c.compute(c.compute_shape({prev->get_shape()}), {prev->eval()});
-
-                    auto l = m.add_literal(r.get_shape(), r.data());
-                    m.replace_instruction(arg, l);
+                    inputs.push_back(arg);
+                    prevs.push_back(prev);
```

Comment:
We really only need to store the `arg` since `prev` can be found using `arg->inputs().front()`.